### PR TITLE
Opening a Thread now correctly marks it as read

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1732

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 1]
```diff
@@ -359,6 +359,11 @@ class ThreadFragment : Fragment() {
                 return@observe
             }
 
+            if (threadState.shouldMarkThreadAsSeen == true) {
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:362]: Yes, that's on purpose. We only mark a Thread as seen when it's opened.

[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:362]: Example of situation where your proposal would be a bad idea :
- Open Thread, read everything, ok.
- Go to Move, Print, or just put the app on background. Or even just scroll the Thread to the top.
- A new mail appears. You don't see it.
- It's marked as seen. You don't know it.
- You go back to ThreadList, you'll never know that you received a new mail.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt [hunk 2]
```diff
@@ -255,13 +255,13 @@ class ThreadViewModel @Inject constructor(
             }
         }
 
-        threadState.shouldMarkThreadAsSeen = thread.unseenMessagesCount > 0
+        if (threadState.shouldMarkThreadAsSeen == null) threadState.shouldMarkThreadAsSeen = thread.unseenMessagesCount > 0
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt:258]: Can't the boolean be non-nullable and the condition be computed right next to where it's used?

**Review by LunarX — CHANGES_REQUESTED** [review]

**Review by LunarX — CHANGES_REQUESTED** [review]
Sometimes my thread is not displayed as "read" when leaving the ThreadFragment but when I open the next thread and come back, the wrongly displayed one is updated and now correctly display as "read"

**Review by LunarX — APPROVED** [review]

**Pull request merged**